src/parsy/__init__.py

```diff
@@ -42,30 +33,37 @@ def __str__(self):
             return f"expected one of {', '.join(expected_list)} at {self.line_info()}"
 
 
-@dataclass
+@dataclass(frozen=True)
+class Position:
+    offset: int
+    line: int
+    column: int
+
+
+@dataclass(frozen=True)
```

Comment:
I made `Result` and `Position` frozen dataclasses, because it seems there's no reason to treat them mutably, and they ought to be hashable if they can be.